File: Ser2023/Company/Company.Web/Modules/Company/Country/CountryForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.Country")]
[BasedOnRow(typeof(CountryRow), CheckNames = true)]
public class CountryForm
{
    [OneThirdWidth]
    public string CountryCode { get; set; }
    [TwoThirdWidth]
    public string CountryName { get; set; }
}

File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/CountryForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface CountryForm {
    CountryCode: StringEditor;
    CountryName: StringEditor;
}

export class CountryForm extends PrefixedContext {
    static formKey = 'Company.Country';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!CountryForm.init)  {
            CountryForm.init = true;

            var w0 = StringEditor;

            initFormType(CountryForm, [
                'CountryCode', w0,
                'CountryName', w0
            ]);
        }
    }
}